addons/mail/static/src/core/web_portal/message_model_patch.js

```diff
@@ -0,0 +1,11 @@
+import { Message } from "@mail/core/common/message_model";
+import { patch } from "@web/core/utils/patch";
+
+patch(Message.prototype, {
+    get canCopyLink() {
+        return false;
+    },
+    canAddReaction(thread) {
+        return super.canAddReaction(thread) && !this.store.self.is_public;
```

Comment:
Public user (with guest) can still add reactions on channels, the condition cannot be generic like that.

Reply:
I'm pretty sure the only way to check this correctly is for the server to return the value based on the actual check that is done on the reaction route.